src/test/groovy/org/prebid/server/functional/tests/AuctionSpec.groovy

```diff
@@ -313,4 +314,36 @@ class AuctionSpec extends BaseSpec {
         and: "BidderRequest and bidRequest ids should be equal"
         assert bidderRequest.id == bidRequestId
     }
+
+    def "PBS should copy imp level passThrough to bidresponse.seatbid[].bid[].ext.prebid.passThrough when the imp.id matches seatbid[].bid[].impid"() {
+        given: "Default bid request with passThrough"
+        def textNode = TextNode.valueOf(PBSUtils.randomString)
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            imp[0].ext.prebid.passThrough = textNode
+        }
+
+        when: "Requesting PBS auction"
+        def response = prebidServerService.sendAuctionRequest(bidRequest)
+
+        then: "BidResponse should contain the same passThrough as on request"
+        assert response.seatbid.first().bid.first().ext.prebid.passThrough == textNode
+
+        and: "BidRequest imp.id should matches with BidResponse seatbid[].bid[].impid"
```

Comment:
I don't see a point in this check. Your test should check for pass-through and not imp id propagation.

Reply:
**Requirement:** Imp-level config: bidrequest.imp.ext.prebid.passthrough --> this object is copied to every instance of bidresponse.seatbid[].bid[].ext.prebid.passthrough **where the imp.id matches seatbid[].bid[].impid.**

We just check that, imp.id and bid[].impid matched by requirements whether is it not necessary to check?

Reply:
This check is pointless in general as is as you have only 1 imp and only 1 bid. If you want to check that requirement, please use >1 imp and >1 bid. And change the wording too, as its a bit misleading right now because I didn't understand how it's related to the feature.